java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -259,21 +418,81 @@ void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
                 }));
             return;
           }
-
         }
         if (instruction.hasReturnValue()) {
           programState = pop.state;
         } else {
           programState = pop.state.stackValue(returnSV);
         }
         break;
-      default:
+      case INVOKEDYNAMIC:
         // do nothing
+        break;
+      case NEW:
+        createNonNullValue(instruction);
+        break;
+      case NEWARRAY:
+      case ANEWARRAY:
+        pop = programState.unstackValue(1);
+        Preconditions.checkState(pop.values.size() == 1, "NEWARRAY needs 1 value on stack");
+        sv = constraintManager.createSymbolicValue(instruction);
+        programState = pop.state.stackValue(sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        break;
+      case ARRAYLENGTH:
+        pop = programState.unstackValue(1);
+        Preconditions.checkState(pop.values.size() == 1, "ARRAYLENGTH needs 1 value on stack");
+        sv = constraintManager.createSymbolicValue(instruction);
+        programState = pop.state.stackValue(sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        break;
+      case ATHROW:
+        pop = programState.unstackValue(1);
+        programState = pop.state.stackValue(constraintManager.createExceptionalSymbolicValue(null));
+        programState.storeExitValue();
+        break;
+      case CHECKCAST:
+        Preconditions.checkState(programState.peekValue() != null, "CHECKCAST needs 1 value on stack");
+        break;
+      case INSTANCEOF:
+        pop = programState.unstackValue(1);
+        Preconditions.checkState(pop.values.size() == 1, "INSTANCEOF needs 1 value on stack");
+        SymbolicValue.InstanceOfSymbolicValue instanceOf = new SymbolicValue.InstanceOfSymbolicValue();
+        instanceOf.computedFrom(pop.valuesAndSymbols);
+        programState = pop.state.stackValue(instanceOf);
+        break;
+      case MONITORENTER:
+      case MONITOREXIT:
+        pop = programState.unstackValue(1);
+        Preconditions.checkState(pop.values.size() == 1, Printer.OPCODES[instruction.opcode] + " needs 1 value on stack");
+        programState = pop.state;
+        break;
+      case MULTIANEWARRAY:
+        BytecodeCFGBuilder.MultiANewArrayInsn multiANewArrayInsn = (BytecodeCFGBuilder.MultiANewArrayInsn) instruction;
+        pop = programState.unstackValue(multiANewArrayInsn.dim);
+        Preconditions.checkState(pop.values.size() == multiANewArrayInsn.dim, "MULTIANEWARRAY needs " + multiANewArrayInsn.dim + " values on stack");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2629)